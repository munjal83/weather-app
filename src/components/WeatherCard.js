import React from 'react';
import Paper from '@material-ui/core/Paper';
import Typography from '@material-ui/core/Typography';

const styles = {
    root: {
        margin: '0 300px 20px 300px',
        padding: 20,
    } 
}

class WeatherCard extends React.Component {


render() {
    return (
        <div>
            <Paper style={styles.root} elevation={3}>
                <Typography variant="display1" component="h3" align="center" gutterBottom>
                    Today's Weather
                        </Typography>
                <Typography variant="title" component="p" align="center" gutterBottom color="secondary">
                    City: {this.props.data.city}
                </Typography>
                <Typography variant="title" component="p" align="center" gutterBottom color="secondary">
                    Last Updated: {this.props.data.lastUpdated}
                </Typography>
                <Typography variant="title" component="p" align="center" gutterBottom color="secondary">
                    Feels Like: {this.props.data.feelsLike}°
                        </Typography>
                <Typography variant="title" component="p" align="center" gutterBottom color="secondary">
                    Temperature Unit: {this.props.data.unit}°
                        </Typography>
                <Typography variant="title" component="p" align="center" gutterBottom color="secondary">
                    Temperature: {this.props.data.temperature}°
                        </Typography>
            </Paper>
        </div>
    );
}
}

export default WeatherCard;