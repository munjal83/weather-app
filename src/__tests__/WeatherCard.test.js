import React from 'react';
import ReactDOM from 'react-dom';
import { create } from 'react-test-renderer';
import WeatherCard from '../components/WeatherCard';

const data = {city: ''}

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<WeatherCard data={data}/>, div);
  ReactDOM.unmountComponentAtNode(div);
});

describe('WeatherCard Component Snapshot',()=>{
  test('testing weathercard component', () => {
  let tree = create(<WeatherCard data={data}/>)
  expect(tree.toJSON()).toMatchSnapshot();
})
});