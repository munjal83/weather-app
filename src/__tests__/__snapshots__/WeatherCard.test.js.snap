// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherCard Component Snapshot testing weathercard component 1`] = `
<div>
  <div
    className="MuiPaper-root-53 MuiPaper-elevation3-58 MuiPaper-rounded-54"
    style={
      Object {
        "margin": "0 300px 20px 300px",
        "padding": 20,
      }
    }
  >
    <h3
      className="MuiTypography-root-80 MuiTypography-display1-84 MuiTypography-gutterBottom-97 MuiTypography-alignCenter-93"
    >
      Today's Weather
    </h3>
    <p
      className="MuiTypography-root-80 MuiTypography-title-86 MuiTypography-colorSecondary-101 MuiTypography-gutterBottom-97 MuiTypography-alignCenter-93"
    >
      City: 
      
    </p>
    <p
      className="MuiTypography-root-80 MuiTypography-title-86 MuiTypography-colorSecondary-101 MuiTypography-gutterBottom-97 MuiTypography-alignCenter-93"
    >
      Last Updated: 
    </p>
    <p
      className="MuiTypography-root-80 MuiTypography-title-86 MuiTypography-colorSecondary-101 MuiTypography-gutterBottom-97 MuiTypography-alignCenter-93"
    >
      Feels Like: 
      °
    </p>
    <p
      className="MuiTypography-root-80 MuiTypography-title-86 MuiTypography-colorSecondary-101 MuiTypography-gutterBottom-97 MuiTypography-alignCenter-93"
    >
      Temperature Unit: 
      °
    </p>
    <p
      className="MuiTypography-root-80 MuiTypography-title-86 MuiTypography-colorSecondary-101 MuiTypography-gutterBottom-97 MuiTypography-alignCenter-93"
    >
      Temperature: 
      °
    </p>
  </div>
</div>
`;
